### PR TITLE
[PATCH v2] Time-related test fixes

### PR DESCRIPTION
**Pull request by lumag**
An attempt to fix https://bugs.linaro.org/show_bug.cgi?id=3987

### CODE REVIEW

File: test/validation/api/scheduler/scheduler.c [hunk 1]
```diff
@@ -185,8 +185,19 @@ static void scheduler_test_wait_time(void)
 	upper_limit = odp_time_local_from_ns(5 * ODP_TIME_SEC_IN_NS +
 							ODP_WAIT_TOLERANCE);
 
-	CU_ASSERT(odp_time_cmp(diff, lower_limit) >= 0);
-	CU_ASSERT(odp_time_cmp(diff, upper_limit) <= 0);
+	if (odp_time_cmp(diff, lower_limit) <= 0) {
```
Comment (Bill-Fischofer-Linaro) [inline test/validation/api/scheduler/scheduler.c]: Shouldn't this be < rather than <= ?

File: test/validation/api/scheduler/scheduler.c [hunk 2]
```diff
@@ -185,8 +185,19 @@ static void scheduler_test_wait_time(void)
...
+		fprintf(stderr, "Exceed lower limit: "
+			"diff is %" PRIu64 ", lower_limit %" PRIu64 "\n",
+			odp_time_to_ns(diff), odp_time_to_ns(lower_limit));
+		CU_FAIL("Exceed lower limit\n");
+	}
+
+	if (odp_time_cmp(diff, upper_limit) >= 0) {
```
Comment (Bill-Fischofer-Linaro) [inline test/validation/api/scheduler/scheduler.c]: Similarly, should be > rather than >= ?

File: test/validation/api/timer/timer.c [hunk 3]
```diff
@@ -334,8 +334,8 @@ static void timer_test_queue_type(odp_queue_type_t queue_type)
 			tim = odp_timeout_timer(tmo);
 			tick = odp_timeout_tick(tmo);
 
-			CU_ASSERT(diff_period > (period_ns - (4 * res_ns)));
-			CU_ASSERT(diff_period < (period_ns + (4 * res_ns)));
+			CU_ASSERT(diff_period > (period_ns - (5 * res_ns)));
+			CU_ASSERT(diff_period < (period_ns + (5 * res_ns)));
```
Comment (Bill-Fischofer-Linaro) [inline test/validation/api/timer/timer.c]: Not sure what the best way to deal with this is, but these sort of ad-hoc "fudge factors" seem awkward.

**Comment by muvarov** [comment]
As we discussed on meeting call validation tests should not test exact time precision. Aim of validation tests is to check that time/timer generally operates and can be used for ODP application. Precision question is for measurement and performance tests and they are depend on platform and execution environment.

From other side, after I wrote lines about we might check if we can use platform capabilities and relay on that numbers instead just increasing boundaries on some numbers.

File: test/validation/api/scheduler/scheduler.c (hunk 1 above)
Comment (lumag) [reply test/validation/api/scheduler/scheduler.c]: It's c&p from timer code. And '<' vs '<=' is 1ns difference.

File: test/validation/api/timer/timer.c (hunk 3 above)
Comment (lumag) [reply test/validation/api/timer/timer.c]: Yes. Moreover they still do not guarantee stable behaviour. I'm playing with the code to fix such kinds of tests.

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**